app/client/templates/elements/eventsTable.js

```diff
@@ -49,8 +49,9 @@ Template['elements_event_table'].helpers({
             searchQuery = TemplateVar.get('search'),
             limit = TemplateVar.get('limit'),
             collection = Events,
-            selector = this.ids ? {_id: {$in: this.ids}} : {};
-
+            selector = this.ids ? {_id: {$in: this.ids.slice(Number((limit+1)*-1))}} : {}; 
```

Comment:
Yes, I've tested it, that's why I multiplied it by -1. See the problem is that the current way you were doing, you were querying 6k ids and then asking it to limit the returning parameters. This created a massive overhead, as you'd never need the 6k ids to begin with.

By slicing the id array I am limiting the amount of ids sent to the selectors so the overhead is considerably less. It's a matter of freezing the whole interface for a full minute versus, instantaneous opening the contract..